modules/hub-acm-feature/main.tf

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  private_key            = var.create_ssh_key && var.ssh_auth_key == null ? tls_private_key.k8sop_creds[0].private_key_pem : var.ssh_auth_key
+  k8sop_creds_secret_key = var.secret_type == "cookiefile" ? "cookie_file" : var.secret_type
+}
+
+resource "google_gke_hub_feature" "acm" {
+  count    = var.enable_fleet_feature ? 1 : 0
+  provider = google-beta
+
+  name     = "configmanagement"
+  project  = var.project_id
+  location = "global"
+}
+
+resource "google_gke_hub_feature_membership" "main" {
+  provider = google-beta
+  depends_on = [
+    google_gke_hub_feature.acm
+  ]
+
+  location = "global"
+  feature  = "configmanagement"
+
+  membership = var.cluster_membership_id
+  project    = var.project_id
+
+  configmanagement {
+    version = "1.11.0"
```

Comment:
Wouldn't this result in possibly older versions being installed? We could default to API with a user override for explicit version.

Reply:
I think it's preferable to be explicit with pinning a version here.

Reply:
Should expose this as a var defaulting to `1.11.0`? I assume users will want to use other versions and it can be toilsome to keep bumping this?

Reply:
We'd still have to keep bumping the default though. A new variable can always be added in a later (minor) release, but for now I think hard-coding is sensible to keep the API surface small.